src/js/services/conversation.js

```diff
@@ -163,12 +144,58 @@ export function resendMessage(messageClientId) {
 
         if (newMessage.type === 'text') {
             return dispatch(sendChain(postSendMessage, newMessage));
+        } else if (newMessage.type === 'location') {
+            if (newMessage.coordinates) {
+                return dispatch(sendChain(postSendMessage, newMessage));
+            } else {
+                return dispatch(sendLocation(newMessage));
+            }
         }
 
         return dispatch(sendChain(postUploadImage, newMessage));
     };
 }
 
+export function sendLocation(message) {
+    return (dispatch, getState) => {
+        const locationMessage = message || dispatch(addMessage(null, {
```

Comment:
I don't know how I feel about passing `null` for the text with extra props when we could just to `addMessage({text: 'something', type: 'location' })` there the object would just merge with the set of defaults in `addMessage`. You could then just omit `text` in there and it would be much cleaner.